### PR TITLE
Implemented Manhattan distance to point.py

### PR DESCRIPTION
**Pull request by jremes-foss**
Manhattan distance implemented to sympy/geometry/point.py.

http://mathworld.wolfram.com/TaxicabMetric.html

Thank you.

### CODE REVIEW

**Comment by asmeurer** [comment]
It looks good, but I'm unsure if this is the name we should use. This is also called the taxicab distance and the l1 distance.  Which name is used most often in 2-dimensional Euclidean geometry?

**Comment by Newman101** [comment]
I think taxicab distance is the most commonly used in the 2D Euclidean Geometry. I can change the name, of course.

**Comment by Newman101** [comment]
That's sorted out.

**Comment by asmeurer** [comment]
I'll see if @smichr has any comments. Otherwise this is +1 from me.

File: sympy/geometry/point.py [hunk 1]
```diff
@@ -269,6 +269,38 @@ def distance(self, p):
         p = Point(p)
         return sqrt(sum([(a - b)**2 for a, b in zip(self.args, p.args)]))
 
+    def taxicab_distance(self, p):
+        """The Taxicab Distance from self to point p.
+
+        http://mathworld.wolfram.com/TaxicabMetric.html
```
Comment (smichr) [inline sympy/geometry/point.py]: I would put this as a reference but not force the SymPy user to go there to know what the function does. The docstring should probably read, "Return the sum of the horizontal and vertical distances to point ``p``."

File: sympy/geometry/point.py [hunk 2]
```diff
@@ -269,6 +269,38 @@ def distance(self, p):
...
+        ========
+
+        >>> from sympy.geometry import Point
+        >>> p1, p2 = Point(1, 1), Point(4, 5)
+        >>> p1.taxicab_distance(p2)
+        7
+
+        """
+        p = Point(p)
+        return sum(abs(a-b) for a,b in zip(self.args, p.args))
```
Comment (smichr) [inline sympy/geometry/point.py]: pep8 nits: -> `sum(abs(a - b) for a, b in zip(self.args, p.args))`

File: sympy/geometry/point.py (hunk 1 above)
Comment (Newman101) [inline sympy/geometry/point.py]: Sure, I can update the docstring.

**Comment by smichr** [comment]
It's in, thanks!

**Pull request merged**